python/cudf/cudf/core/frame.py

```diff
@@ -573,6 +582,18 @@ def equals(self, other, **kwargs):
         else:
             return self._index.equals(other._index)
 
+    def _explode(self, explode_column_num: int, index: Optional[cudf.Index]):
+        if index is not None:
+            explode_column_num += index.nlevels
+        res_tbl = libcudf.lists.explode_outer(
+            cudf._lib.table.Table(self._data, index=index), explode_column_num
```

Comment:
It feels a bit awkward that we create a new `Table` here instead of using the existing table that the `Frame` already has.